docs/public/specifications/tests/Connectivity-TP/connectivity04.md

```diff
@@ -0,0 +1,235 @@
+   2. Add found IP prefix, if any, with *NS IP Address* to the *IPv4 Prefix*
+      and *IPv6 Prefix* sets, respectively.
+
+5. If the *IPv4 Prefix* set is non-empty, then do:
+   1. For each IP prefix in the set that has two or more members, output
+      *[CN04_IPV4_SAME_PREFIX]* with the prefix and list of all members for that prefix.
+   2. For each IP prefix in the set that has exactly one member, output
+      *[CN04_IPV4_DIFFERENT_PREFIX]* with the combined set of their associated members.
+
+6. If the *IPv6 Prefix* set is non-empty, then do:
+   1. For each IP prefix in the set that has two or more members, output
+      *[CN04_IPV6_SAME_PREFIX]* with the prefix and list of all members for that prefix.
+   2. For each IP prefix in the set that has exactly one member, output
+      *[CN04_IPV6_DIFFERENT_PREFIX]* with the combined set of their associated members.
+
+## Outcome(s)
+
+The outcome of this Test Case is "fail" if there is at least one message
+with the severity level *[ERROR]* or *[CRITICAL]*.
+
+The outcome of this Test Case is "warning" if there is at least one message
+with the severity level *[WARNING]*, but no message with severity level
+*ERROR* or *CRITICAL*.
+
+In other cases, no message or only messages with severity level
+*[INFO]* or *[NOTICE]*, the outcome of this Test Case is "pass".
+
+## Special procedural requirements
+
+This Test Case is dependent on one of two possible services that can provide
+ASN lookup (Cymru or RIPE RIS). The service must be available over the network.
+
+The *Child Zone* must be a valid name meeting
+"[Requirements and normalization of domain names in input]".
+
+### Cymru prefix lookup
+
+The Cymru prefix lookup is described on the Team Cymru [IP to ASN Mapping]
+using DNS lookup.
+
+1. Prepend the *Cymru Base Name* with the label "origin" (IPv4) or 
+   "origin6" (IPv6) ("Expanded Base Name"). Example of expanded basenames :
+   
+```
+origin.asnlookup.zonemaster.net
+origin6.asnlookup.zonemaster.net
+```
+
+2. Reverse the IP address with the same method as is used for
+   reverse lookup ("Reverse IP"). For description see [RFC 1035][RFC 1035#3.5],
+   section 3.5, for IPv4 and [RFC 3596][RFC 3596#2.5], section 2.5, for IPv6.
+ 
+3. Prepend the *Expanded Base Name* with *Reverse IP* ("Query Name").
+   See [IP to ASN Mapping] for details.
+
+4. Create a [DNS Query] with query type TXT and query name *Query Name*.
+   ("TXT Query").
+
+5. [Send] *TXT Query* to the *Cymru Base Name*.
+
+6. If at least one of the following criteria is met, output
+   *[CN04_EMPTY_PREFIX_SET]* and exit this lookup:
+   1. The [DNS Response] has the [RCODE Name] NXDomain.
+   2. The [DNS Response] has the [RCODE Name] NoError and an empty answer section.
+
+7. If at least one of the following criteria is met, output
+   *[CN04_ERROR_PREFIX_DATABASE]* and exit this lookup:
+   1. There is no response.
+   2. The [DNS Response] does not have the [RCODE Name] NoError.
```

Comment:
It should rather be "NoError or NXDomain".

Reply:
True, but repeating “NoError or NXDomain” would make the intent clearer.